chain/cosmos/client.go

```diff
@@ -0,0 +1,152 @@
+package cosmos
+
+import (
+	"context"
+	"encoding/hex"
+	"fmt"
+	"time"
+
+	"github.com/renproject/multichain/api/account"
+	"github.com/renproject/pack"
+
+	cliContext "github.com/cosmos/cosmos-sdk/client/context"
+	"github.com/cosmos/cosmos-sdk/codec"
+	"github.com/cosmos/cosmos-sdk/x/auth"
+	"github.com/cosmos/cosmos-sdk/x/auth/client/utils"
+	rpchttp "github.com/tendermint/tendermint/rpc/client/http"
+)
+
+const (
+	// DefaultClientTimeout used by the Client.
+	DefaultClientTimeout = time.Minute
+	// DefaultClientTimeoutRetry used by the Client.
+	DefaultClientTimeoutRetry = time.Second
+	// DefaultClientHost used by the Client. This should only be used for local
+	// deployments of the multichain.
+	DefaultClientHost = "http://0.0.0.0:26657"
+	// DefaultBroadcastMode configures the behaviour of a cosmos client while it
+	// interacts with the cosmos node. Allowed broadcast modes can be async, sync
+	// and block. "async" returns immediately after broadcasting, "sync" returns
+	// after the transaction has been checked and "block" waits until the
+	// transaction is committed to the chain.
+	DefaultBroadcastMode = "sync"
+)
+
+// ClientOptions are used to parameterise the behaviour of the Client.
+type ClientOptions struct {
+	Timeout       time.Duration
+	TimeoutRetry  time.Duration
+	Host          pack.String
+	BroadcastMode pack.String
+}
+
+// DefaultClientOptions returns ClientOptions with the default settings. These
+// settings are valid for use with the default local deployment of the
+// multichain. In production, the host, user, and password should be changed.
+func DefaultClientOptions() ClientOptions {
+	return ClientOptions{
+		Timeout:       DefaultClientTimeout,
+		TimeoutRetry:  DefaultClientTimeoutRetry,
+		Host:          pack.String(DefaultClientHost),
+		BroadcastMode: pack.String(DefaultBroadcastMode),
+	}
+}
+
+// WithHost sets the URL of the Bitcoin node.
+func (opts ClientOptions) WithHost(host pack.String) ClientOptions {
+	opts.Host = host
+	return opts
+}
+
+// A Client interacts with an instance of the Cosmos based network using the REST
+// interface exposed by a lightclient node.
+type Client interface {
+	// Account query account with address
+	Account(address Address) (Account, error)
+
+	// Client interface from account.Client
+	account.Client
+}
+
+type client struct {
+	opts   ClientOptions
+	cliCtx cliContext.CLIContext
+}
```

Comment:
```suggestion
type Client struct {
	opts   ClientOptions
	cliCtx cliContext.CLIContext
}
```
Instead of creating an interface, you can just expose the actual concrete implementation.